mne/io/eeglab/eeglab.py

```diff
@@ -609,7 +609,12 @@ def _read_annotations_eeglab(eeg, uint16_codec=None):
     onset = [event.latency - 1 for event in events]
     duration = np.zeros(len(onset))
     if len(events) > 0 and hasattr(events[0], 'duration'):
-        duration[:] = [event.duration for event in events]
+        for idx, event in enumerate(events):
+            if isinstance(events[idx].duration, np.ndarray):
```

Comment:
You could replace `events[idx].duration` with `event.duration`.

Also, why do you set `this_duration` to `np.nan` if `event.duration` is an ndarray (regardless of its contents)?

Reply:
> You could replace events[idx].duration with event.duration.

Yes, I have that waiting locally, I know its ugly. :)

> Also, why do you set this_duration to np.nan if event.duration is an ndarray (regardless of its contents)?

Good question - I don't know what is the default in our case. I can set it to 0 as well. I think I was biased by the contents of this file - some of the durations were `NaN` in this case. And reading works when the durations are NaN.
I don't remember how common are nan durations in eeglab, maybe that's something that changed in more recent versions.

Reply:
I was more referring to the fact that you assume a missing duration if the type of `event.duration` is an ndarray. Why is that? What is the value of the `ndarray` in this case and wouldn't it make more sense to condition on the value?

Reply:
the array is empty in this case, I don't know if there are circumstances when the duration is an array but is not empty. My guess is that this behavior is specific to reading matlab structures with loadmat - structures can have empty fields which are read as empty arrays. I can add a check to make sure that `len(event.duration) == 0`.

Reply:
Yes, I think a check would be good. So this means that non-missing durations are never stored as an ndarray but a float?